docs/06-contributors/999-rfcs/2819-stdlib.md

```diff
@@ -0,0 +1,115 @@
+# #2819 - Wing Utility Library
+
+- **Author(s):**: @eladb
+- **Submission Date**: 2023-06-07
+- **Stage**: Proposal
+- **Stage Date**: 2023-06-07
+
+> A specification for the Wing SDK utility libraries.
+
+## Design
+
+The Wing SDK includes a set of submodules which offer a set of useful functionality for building
+cloud applications.
+
+> This is just an initial spec and will likely change in the future.
+
+For each module, we show a list of function declarations. For each function, we specify which phase this function
+is bound to: `inflight` (inflight only - async), `preflight` (preflight only - synchronous) or
+`preflight inflight` (for phase-independent).
+
+## `util` module
+
+```js
+// read environment variables
+preflight inflight env(name: str): str;
+preflight inflight tryEnv(name: str): str?;
+
+// sleep
+inflight sleep(d: duration): void; // inflight only!
+
+// hash/random/uid generation
+preflight inflight uuidv4(): str;
+preflight inflight sha256(data: str): str;
+preflight inflight nanoid(): str; // https://www.npmjs.com/package/nanoid
+
+// returns an pseudo random integer random number between [0, max] (inclusive).
+// the default `max` is JavaScript's `Number.MAX_VALUE`.
+preflight inflight random(max: num?): num;
+
+preflight inflight class RegExp {
+  test(s: str): bool;
+  match(s: str): RegExpMatch;
+}
+
+preflight inflight regex(pattern: str): RegExp;
+```
+
+### `fs` module
+
+File APIs are phase-independent and at the moment are all implemented using the "sync" node.js `fs` APIs.
+In the future we might separate preflight/inflight versions and the inflight versions will be async.
```

Comment:
Maybe we want to say a word about how these operations work on underlying compute platforms. For example, do these APIs assume all paths are UNIX based? Or fallback to the definition / spec according to Node.js or WASI? I wonder what's the most pragmatic thing to do here...

Reply:
Added a note that all paths are POSIX paths (`/` instead of `\`).